pkg/handler/middleware/middleware.go

```diff
@@ -883,3 +886,45 @@ func getPrivilegedMLAAdminSettingProvider(ctx context.Context, clusterProviderGe
 
 	return mlaAdminSettingProviderGetter(seed)
 }
+
+// PrivilegedIPAMPool is a middleware that injects the current PrivilegedIPAMPoolProvider into the ctx.
+func PrivilegedIPAMPool(clusterProviderGetter provider.ClusterProviderGetter, ipamPoolProviderGetter provider.PrivilegedIPAMPoolProviderGetter, seedsGetter provider.SeedsGetter) endpoint.Middleware {
+	return func(next endpoint.Endpoint) endpoint.Endpoint {
+		return func(ctx context.Context, request interface{}) (response interface{}, err error) {
+			seedCluster := request.(seedClusterGetter).GetSeedCluster()
+			privilegedIPAMPoolProvider, err := getPrivilegedIPAMPoolProvider(ctx, clusterProviderGetter, ipamPoolProviderGetter, seedsGetter, seedCluster.SeedName, seedCluster.ClusterID)
+			if err != nil {
+				return nil, err
+			}
+			ctx = context.WithValue(ctx, PrivilegedIPAMPoolProviderContextKey, privilegedIPAMPoolProvider)
+			return next(ctx, request)
+		}
+	}
+}
+
+func getPrivilegedIPAMPoolProvider(ctx context.Context, clusterProviderGetter provider.ClusterProviderGetter, ipamPoolProviderGetter provider.PrivilegedIPAMPoolProviderGetter, seedsGetter provider.SeedsGetter, seedName, clusterID string) (provider.PrivilegedIPAMPoolProvider, error) {
+	seeds, err := seedsGetter()
+	if err != nil {
+		return nil, err
+	}
+
+	if clusterID != "" {
```

Comment:
the clusterID will always be empty so no point to have it in any of these PrivilegedIPAM.. functions.

Check the BackupCredentials for inspiration, they are also a Seed Resource